chore: fix buf v2 linting errors by updating proto packages

syedazeez337 opened the pull request:
This change removes the buf v2 lint exceptions and fixes all linting errors:
- Package directory structure now matches package names
- Package names have version suffix (.v1)
- Service names have Service suffix
- Updated all Go imports to new paths

This is a breaking change for external extension developers who import the proto packages.

Fixes #7834

syedazeez337 committed:
chore: fix buf v2 linting errors by updating proto packages

This change removes the buf v2 lint exceptions and fixes all linting errors:
- Package directory structure now matches package names
- Package names have version suffix (.v1)
- Service names have Service suffix
- Updated all Go imports to new paths
- Fix import ordering to comply with goimports/gci linting rules

This is a breaking change for external extension developers who import
the proto packages.

Fixes #7834

Signed-off-by: Azeez Syed <syedazeez337@gmail.com>

jukie committed:
Merge branch 'main' into fix/buf-v2-lint-errors

syedazeez337 commented on examples/simple-extension-server/go.mod:
```diff
@@ -20,7 +20,9 @@ require (
 	google.golang.org/protobuf v1.36.11 // indirect
 )
+
+replace github.com/envoyproxy/gateway => ../../
```
The simple-extension-server example imports github.com/envoyproxy/gateway/envoygateway/extension/v1, which doesn't exist in the released module (v1.6.1) yet. The replace directive allows the example to use the local repo's new paths. This is necessary until this PR is merged and a new version is released.
Without this I cannot test this locally

jukie replied:
Great, thanks!

syedazeez337 committed:
fix(examples): update simple-extension-server Docker build for replace directive

Update Dockerfile and Makefile to build from repo root context so the
replace directive (../../) can resolve the gateway module correctly
during Docker builds.

Signed-off-by: Syed Azeez <syedazeez337@gmail.com>
Signed-off-by: Azeez Syed <syedazeez337@gmail.com>

zhaohuabing commented:
Can we avoid changing the package name and service name? This would break existing extension servers, so it'd be great if we could keep them as-is if possible.

syedazeez337 committed:
fix(examples): revert simple-extension-server to use published proto paths

Revert simple-extension-server example to use the old proto import paths
from the published v1.6.1 release. This removes the replace directive that
was breaking CI and addresses reviewer feedback.

Changes:
- Revert Dockerfile and Makefile to original build context
- Remove replace directive from go.mod
- Revert import path to proto/extension (from published release)
- Add README.md to nighthawk/client/v1 to preserve documentation

The simple-extension-server will be updated to use new proto paths in a
follow-up PR after a new release is published.

Signed-off-by: Azeez Syed <syedazeez337@gmail.com>

syedazeez337 commented:
Hi @zirain @zhaohuabing 
https://github.com/envoyproxy/gateway/issues/7834
this was warned it will contain breaking changes. I suppose that was intentional.
@jukie I have update my PR with scope of only addressing the Proto changes. Let me know how this goes

syedazeez337 commented:
Let me know about this community meeting and if I can part of the discussion.

cnvergence commented:
@syedazeez337 
Absolutely, feel free to join, we have two timeslots
https://github.com/envoyproxy/gateway?tab=readme-ov-file#community-meeting

syedazeez337 commented:
Hi @cnvergence 
Sorry to bother you, it was not clear when I was adding the dates to my calender, one says 11th Dec and another 18th Dec, do we get real time updates? which timezone is tomorrow's meeting?
